src/switch_exporter/switch.py

```diff
@@ -68,7 +69,8 @@ async def _connect(self) -> None:
     async def _connect_unlocked(self) -> None:
         self.conn = await asyncssh.connect(
             self.hostname, known_hosts=None,
-            username=self.username, password=self.password)
+            username=self.username, password=self.password,
+            client_keys=self.keyfile, signature_algs=['rsa-sha2-512'])
```

Comment:
Restricting the set of signature algorithms does seem worrying. According to the asyncssh docs, this is already one of the supported algorithms.

My guess is that because `self.keyfile` is hardcoded as `[]`, asyncssh defaults to using the first private key it finds in your ~/.ssh, and that particular key is not authorized on the switch, while specifying `signature_algs` forces it to instead find your rsa-sha2-512 key. Once you've wired up the command-line argument to `self.keyfile`, see if you can remove `signature_algs`.

Reply:
I've fixed the variable usage, but this issue is still unfortunately present. Looking at my debug, it seems asyncssh is sending the key with a  rsa-sha2-256 signature, which is being rejected by the Mellanox server. What I could do is add a commandline flag to specific this.

Reply:
Alternately, we could use the ssh config. Might be a tidier solution

Reply:
I'll give this a test with our switches when I get a chance. For comparison, what version of Onyx are you running? And what version of the asyncssh and cryptography packages?

Reply:
I found the cause. If `Server security strict mode` is turned on, then this breaks because the server advises the wrong supported signature_algs, and AsyncSSH uses rsa-sha2-256 as default. We will have to fix this on our end somehow, but we could remove that variable from my branch for bringing in to the repo

Reply:
With the same versions of cryptography and asyncssh but with Onyx 3.10.4408 I'm not able to reproduce the need for `signature_algs`. And it looks like the strict security mode is enabled:
```
# show ssh server
SSH server configuration:
   SSH server enabled:             yes
   Server security strict mode:    yes
```

Reply:
I will update my firmware I guess!